src/OpenTelemetry.AutoInstrumentation.NLogTarget/OpenTelemetryTarget.cs

```diff
@@ -0,0 +1,211 @@
+
+    public int ScheduledDelayMilliseconds { get; set; } = 5000;
+
+    public int MaxQueueSize { get; set; } = 2048;
+
+    public int MaxExportBatchSize { get; set; } = 512;
+
+    protected override void InitializeTarget()
+    {
+        base.InitializeTarget();
+
+        if (_loggerProvider != null)
+        {
+            return;
+        }
+
+        var createLoggerProviderBuilderMethod = typeof(Sdk).GetMethod("CreateLoggerProviderBuilder", System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.NonPublic)!;
+        var loggerProviderBuilder = (LoggerProviderBuilder)createLoggerProviderBuilderMethod.Invoke(null, null)!;
+
+        loggerProviderBuilder = loggerProviderBuilder
+            .SetResourceBuilder(ResourceConfigurator.CreateResourceBuilder(Instrumentation.GeneralSettings.Value.EnabledResourceDetectors));
+
+        loggerProviderBuilder = loggerProviderBuilder.AddOtlpExporter(options =>
+        {
+            var endpoint = Endpoint;
+            if (string.IsNullOrEmpty(endpoint))
+            {
+                endpoint = Environment.GetEnvironmentVariable("OTEL_EXPORTER_OTLP_ENDPOINT");
+            }
+
+            if (!string.IsNullOrEmpty(endpoint))
+            {
+                options.Endpoint = new Uri(endpoint!, UriKind.RelativeOrAbsolute);
+            }
+
+            if (!string.IsNullOrEmpty(Headers))
+            {
+                options.Headers = Headers;
+            }
+
+            options.Protocol = UseHttp ? OpenTelemetry.Exporter.OtlpExportProtocol.HttpProtobuf : OpenTelemetry.Exporter.OtlpExportProtocol.Grpc;
+            options.BatchExportProcessorOptions.ScheduledDelayMilliseconds = ScheduledDelayMilliseconds;
+            options.BatchExportProcessorOptions.MaxQueueSize = MaxQueueSize;
+            options.BatchExportProcessorOptions.MaxExportBatchSize = MaxExportBatchSize;
+        });
+
+        _loggerProvider = loggerProviderBuilder.Build();
+        _getLoggerFactory = CreateGetLoggerDelegate(_loggerProvider);
+    }
+
+    protected override void Write(LogEventInfo logEvent)
+    {
+        if (_loggerProvider is null)
+        {
+            return;
+        }
+
+        if (Sdk.SuppressInstrumentation)
+        {
+            return;
+        }
+
+        var logger = GetOrCreateLogger(logEvent.LoggerName);
+
+        // Build properties from event properties and context
+        var properties = new List<KeyValuePair<string, object?>>();
+        if (IncludeEventProperties && logEvent.HasProperties && logEvent.Properties is not null)
+        {
+            foreach (var kvp in logEvent.Properties)
+            {
+                properties.Add(new KeyValuePair<string, object?>(Convert.ToString(kvp.Key)!, kvp.Value));
+            }
+        }
+
+        // Scope properties can be added via explicit <attribute> entries or NLog's contexts (GDC/MDLC)
+        foreach (var attribute in Attributes)
+        {
+            var value = attribute.Layout?.Render(logEvent);
```

Comment:
instead use the following
```c#
var value = RenderLogEvent(attribute.Layout, logEvent);
```

Reply:
resolved in [4401757](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/4371/commits/440175748c7fc3d7ee52f84af92a7d813686ed9d)